File: backend/parking-domain/src/main/java/com/chatchatabc/parking/domain/model/Role.java
package com.chatchatabc.parking.domain.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Entity
@Table(name = "roles")
@AllArgsConstructor
@NoArgsConstructor
public class Role {
    @JsonIgnore
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(unique = true)
    private String name;

    @JsonIgnore
    @ManyToMany(mappedBy = "roles")
    private List<User> users = List.of();

    /**
     * Default constructor
     *
     * @param name Role name
     */
    public Role(String name) {
        this.name = name;
    }
}

File: backend/parking-service/src/main/kotlin/com/chatchatabc/service/domain/model/Role.kt
package com.chatchatabc.service.domain.model

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import lombok.Data

@Data
@Entity
@Table(name = "roles")
open class Role(name: String) {
    @JsonIgnore
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    open lateinit var id: String

    @Column(unique = true)
    open var name: String = name

    @JsonIgnore
    @ManyToMany(mappedBy = "roles")
    open var users: MutableList<User> = mutableListOf()
}